sherpa/astro/data.py

```diff
@@ -403,6 +403,19 @@ def __setstate__(self, state):
             self.header = None
         self.__dict__.update(state)
 
+    def _validate(self, name, energy_lo, energy_hi, ethresh):
+        """
+        Default validation: do nothing and just pass the arrays through.
+        Subclasses may override this method to perform different validations.
+
+        :param name: The name/label of the current file
+        :param energy_lo: The lower bounds of the energy bins
+        :param energy_hi: The upper bounds of the energy bins
+        :param ethresh: The lowest energy value, if any
+        :return: A tuple of energy bounds arrays, validated and possibly adjusted
+        """
```

Comment:
We've been using something like the following (although I am never sure what textual label to use for NumPy array here):

    Parameters
    ----------
    name : str
        The name/label of the current file.
    energ_lo, energ_hi : NumPy array
        The lower and upper bounds of the energy bins. The arrays must have
         the same size
    ethresh : float
        The lowest valid energy value.

    Returns
    -------
    energ_lo, energ_hi : NumPy array
        The energy values to use for the bin boundaries

Reply:
Right. I thought I had already configured my IDE to use the right style. I'll check again.